.circleci/config.yml

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright (c) 2021, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+version: 2.1
+
+executors:
+  daml-docker-executor:
+    docker:
+      - image: cimg/openjdk:11.0
+
+commands:
+  install_sdk:
+    description: "Install Daml SDK"
+    steps:
+      - run:
+          name: Install Daml SDK
+          command: |
+            curl -sSL https://get.daml.com/ | sh /dev/stdin ${SDK_VERSION}
+            # idea from https://circleci.com/docs/2.0/env-vars/
+            >> $BASH_ENV echo 'export PATH="$HOME/.daml/bin:$PATH"'
+  install_yq:
+    description: "Install yq from binary"
+    steps:
+      - run:
+          name: Install yq
+          command: |
+            mkdir -p $HOME/yq
+            curl -L https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64 -o $HOME/yq/yq &&\
+              chmod +x $HOME/yq/yq &&\
+              echo 'export PATH=$HOME/yq/yq:$PATH' >> $BASH_ENV
+  install_gh:
+    description: "Install GitHub Cli"
+    steps:
+      - run:
+          name: Install gh
+          command: |
+            curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+            echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+            sudo apt update
```

Comment:
Won't `apt update` make this non-deterministic? by that I mean that depending on _when_ you run this script, the installed `gh` may be a different version?

Reply:
`apt update` doesn't make it non-determinstic - its required to update the Ubuntu package manager with the added package information for the `GitHub Cli` at line 40.

Here's what you get if you don't run the `apt update` as part of the installation commands :

```
circleci@24776222d1b7:~/project$ curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
3+1 records in
3+1 records out
1795 bytes (1.8 kB, 1.8 KiB) copied, 0.0605322 s, 29.7 kB/s
circleci@24776222d1b7:~/project$ echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
circleci@24776222d1b7:~/project$ sudo apt install gh
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package gh
```

I'd prefer to always install the latest version for security purposes. Happy to discuss.

Reply:
> But I suspect it doesn't just refresh the github client information, but all packages. But I guess, as we're not installing them (just updating the metadata), it should be ok - just need to make sure we don't install other packages in the future. Otherwise the behaviour will differ depending on when you run the CI.

`apt-update` is required when running Ubuntu as Docker image for most cases as these images are kept as minimal as possible in order to save space. Therefore, the Docker image creator leaves this up to the developer using their images to run this command as needed. See here https://github.com/tianon/docker-brew-ubuntu-core/blob/9c78896fe39d5be95cc4a6ae6c0475599a1bc396/xenial/Dockerfile#L36. 

> I would expect that the versions should be fixed in the initial container you select (openjdk11 or whatever), and that this is updated and vetted with appropriate package versions by CircleCI. But I could be entirely wrong.

I'm referring here only to the installation of `Github cli` as its making HTTPS/SSH requests against the GitHub API. I could make a curl command to the correct Github API (here https://docs.github.com/en/rest/reference/releases#create-a-release) avoid using the Github cli tool, but the tool works better with scripting and when dealing with HTTP responses.